### PR TITLE
Add support for conmonrs

### PR DESCRIPTION
**Pull request by haircommander**
#### What type of PR is this?

/kind feature
#### What this PR does / why we need it:
This PR currently carries a couple of other ones including https://github.com/cri-o/cri-o/pull/5823 and https://github.com/cri-o/cri-o/pull/5832
#### Does this PR introduce a user-facing change?

```release-note
add experimental support for conmonrs, a redesign of conmon in rust. It should not be used in production.
```

### CODE REVIEW

**Comment by haircommander** [comment]
besides vendoring and the two aforementioned PRs, the only piece that should affect the rest of cri-o is https://github.com/cri-o/cri-o/pull/5839/commits/a694f5abfc77f69b572de2dcb9de05597462bc98. It's a pretty low risk commit. Thus, the whole PR should not incur too much risk.

Also note: all integration tests are not currently passing. there are a couple of intentionally dropped features (internal log rotation is one of them, in favor of the kubelet's). I am investigating the rest

Ultimately, since we mark this as experimental, and the overall impact to the rest of crio is very low, this is ready for review

**Comment by haircommander** [comment]
oh right it currently relies on https://github.com/containers/conmon-rs/pull/365 too

File: internal/oci/runtime_pod.go [hunk 1]
```diff
@@ -0,0 +1,230 @@
...
+	"github.com/cri-o/cri-o/pkg/config"
+	rspec "github.com/opencontainers/runtime-spec/specs-go"
+	"github.com/pkg/errors"
+	"github.com/sirupsen/logrus"
+	"golang.org/x/net/context"
+	"k8s.io/client-go/tools/remotecommand"
+	types "k8s.io/cri-api/pkg/apis/runtime/v1"
+)
+
+var errNotImplemented = fmt.Errorf("not implemented")
```
Comment (saschagrunert) [inline internal/oci/runtime_pod.go:21]:
```suggestion
var errNotImplemented = errors.New("not implemented")
```

File: internal/oci/runtime_pod.go [hunk 2]
```diff
@@ -0,0 +1,230 @@
...
+
+// runtimeOCI is the Runtime interface implementation relying on conmon-rs to
```
Comment (saschagrunert) [inline internal/oci/runtime_pod.go:23]:
```suggestion
// runtimePod is the Runtime interface implementation relying on conmon-rs to
```

File: internal/oci/runtime_pod.go [hunk 3]
```diff
@@ -0,0 +1,230 @@
...
+// newRuntimePod creates a new runtimePod instance
+func newRuntimePod(r *Runtime, handler *config.RuntimeHandler, c *Container) (RuntimeImpl, error) {
+	// If the container is not an infra container, use the client of the infra container.
+	if !c.IsInfra() {
+		// Search for the runtimePod instance belonging to the sandbox.
+		impl, ok := r.runtimeImplMap[c.Sandbox()]
+		if !ok {
+			// This is a programming error.
+			// This code block assumes every container has a conmon client created already.
+			panic(fmt.Errorf("Attempted to create a new runtime without creating a pod first"))
```
Comment (saschagrunert) [inline internal/oci/runtime_pod.go:43]:
```suggestion
			panic("Attempted to create a new runtime without creating a pod first")
```

**Comment by saschagrunert** [comment]
The GitHub actions CI is now green, but the RPMs as well as kata does not build, because we need go 1.18 for that.

@fidencio do you think we can switch to go 1.18 for the kata runs?

**Comment by fgiudici** [comment]
> The GitHub actions CI is now green, but the RPMs as well as kata does not build, because we need go 1.18 for that.
> 
> @fidencio do you think we can switch to go 1.18 for the kata runs?

I will take a look later today (CC: @littlejawa)

**Comment by haircommander** [comment]
WDYT about adding skip pod runtime build flag to contrib/test/city/cri-o.spec so it's skipped for openshift tests for now (will need to do the same downstream)

**Comment by haircommander** [comment]
blocked on https://github.com/kata-containers/tests/pull/4768

**Comment by fidencio** [comment]
> The GitHub actions CI is now green, but the RPMs as well as kata does not build, because we need go 1.18 for that.
> 
> @fidencio do you think we can switch to go 1.18 for the kata runs?

@saschagrunert, @haircommander, please, send me a direct message on Slack on situations like this as I'm severely lacking behind on my GitHub notifications.

I've raised and merged https://github.com/kata-containers/tests/pull/4768, which should be enough to solve this issue, and I will ensure we have this sorted out **Today**. :-)

Sorry for the non intentional ghosting.

**Comment by fidencio** [comment]
> blocked on [kata-containers/tests#4768](https://github.com/kata-containers/tests/pull/4768)

not anymore, the CI is green. :-)

**Review by saschagrunert — APPROVED** [review]
/lgtm

**Pull request merged**